### PR TITLE
Add extra checks to stop errors that aren't required

### PR DESCRIPTION
**Pull request by HarryET**
# Description

Added extra check and an exception.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] Ran `dartfmt -w -l 120 --fix .`
- [ ] Ran `dartanalyzer --options analysis_options.yaml .`
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: nyxx_interactions/lib/src/events/InteractionEvent.dart [hunk 1]
```diff
@@ -20,13 +20,17 @@ abstract class InteractionEvent<T extends Interaction> {
...
-    final response = await this._client.httpEndpoints.sendRawRequest(url, "POST", body: body);
+      final response = await this._client.httpEndpoints.sendRawRequest(url, "POST", body: body);
 
-    if (response is HttpResponseError) {
-      return Future.error(response);
+      if (response is HttpResponseError) {
+        return Future.error(response);
+      }
+    } else {
+      return Future.error(ResponseRequiredError());
```
Comment (l7ssha) [inline nyxx_interactions/lib/src/events/InteractionEvent.dart]: This check can be inverted
```
if not responded:
  throw

rest of the logic
```

if will remove unecessary if and else

File: nyxx_interactions/lib/src/exceptions/ResponseRequired.dart [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+/// Thrown when you haven't sent a response yet
+class ResponseRequiredError implements Error {
+  /// Returns a string representation of this object.
+  @override
+  String toString() => "ResponseRequiredError: Interaction needs an initial response before followups can be sent.";
+
+  @override
+  StackTrace? get stackTrace => StackTrace.empty;
+}
```
Comment (l7ssha) [inline nyxx_interactions/lib/src/exceptions/ResponseRequired.dart]: no empty line at the end of file

**Review by l7ssha — CHANGES_REQUESTED** [review]

**Comment by l7ssha** [comment]
Please revert also dartfmt commit

**Comment by HarryET** [comment]
Reverted dartfmt @l7ssha

**Pull request merged**